src/integrationTests/autoRefresh.test.ts

```diff
@@ -48,11 +53,17 @@ describe("when auto refreshing a non-expired identity which does not require a r
   });
   test("should set refresh timer", () => {
     expect(setTimeout).toHaveBeenCalledTimes(1);
-    expect(clearTimeout).not.toHaveBeenCalled();
+    expect(clearTimeout).toHaveBeenCalledTimes(1);
```

Comment:
This doesn't seem right - before it was saying that the timeout should be set and not cancelled (i.e. we know it will run). Now it's saying the timeout should be set and cancelled. I actually think this is uncovering a bug in the `beforeEach` method, and one of the other tests was always wrong (the "should not invoke the callback" test just above this - it should invoke the callback because it's got a valid identity the callback could use). Let me know if you want to pair on this one.

Reply:
I found the cause of this issue is that `this._refreshTimerId` is not get reset after the function gets executed, I've now add `this._refreshTimerId = null;` to the end of the setTimeout function

Reply:
Good catch - that fixes a possible bug. But this test looks wrong - after you init with a token which doesn't need to be refreshed, you expect setTimeout to be called once, but clearTimeout shouldn't be called. Or if the timer has already run once, and it's set a new timer, then setTimeout should have been called twice, but there's still no reason it should call clearTimeout.

So why is the test saying that clearTimeout should have been called?